src/main/java/com/mraof/minestuck/client/gui/StatStorerScreen.java

```diff
@@ -0,0 +1,95 @@
+
+public class StatStorerScreen extends Screen
+{
+	private static final ResourceLocation guiBackground = new ResourceLocation("minestuck", "textures/gui/generic_medium.png");
+	
+	private static final int guiWidth = 150;
+	private static final int guiHeight = 98;
+	
+	private static final String divideValueMessage = "Divide power output by:";
+	
+	StatStorerTileEntity te;
+	private StatStorerTileEntity.ActiveType activeType;
+	private static int divideValueBy;
+	
+	public Button typeButton;
+	
+	private TextFieldWidget divideTextField;
+	
+	
+	StatStorerScreen(StatStorerTileEntity te)
+	{
+		super(new StringTextComponent("Stat Storer"));
+		
+		this.te = te;
+		this.activeType = te.getActiveType();
+	}
+	
+	@Override
+	public void init()
+	{
+		addButton(typeButton = new ExtendedButton(this.width / 2 - 67, (height - guiHeight) / 2 + 15, 135, 20, new StringTextComponent(activeType.getNameNoSpaces()), button -> changeActiveType()));
+		int yOffset = (this.height / 2) - (guiHeight / 2);
+		this.divideTextField = new TextFieldWidget(this.font, this.width / 2 - 18, yOffset + 50, 40, 18, new StringTextComponent("Divide comparator output strength"));	//TODO Use translation instead, and maybe look at other text fields for what the text should be
+		this.divideTextField.setValue(String.valueOf(te.getDivideValueBy()));
+		addButton(divideTextField);
+		setInitialFocus(divideTextField);
+		
+		addButton(new ExtendedButton(this.width / 2 - 18, yOffset + 70, 40, 20, new StringTextComponent("DONE"), button -> finish()));
+	}
+	
+	/**
+	 * Gets the ordinal of the current active type, and either cycles it back to the beginning at one if it is the last ordinal in the set or just sets it to the next ordinal in line
+	 */
+	private void changeActiveType()
+	{
+		activeType = StatStorerTileEntity.ActiveType.fromInt(activeType.ordinal() < StatStorerTileEntity.ActiveType.values().length - 1 ? activeType.ordinal() + 1 : 0);
+		typeButton.setMessage(new StringTextComponent(activeType.getNameNoSpaces()));
+	}
+	
+	@Override
+	public void render(MatrixStack matrixStack, int mouseX, int mouseY, float partialTicks)
+	{
+		this.renderBackground(matrixStack);
+		RenderSystem.color4f(1F, 1F, 1F, 1F);
+		this.minecraft.getTextureManager().bind(guiBackground);
+		int yOffset = (this.height / 2) - (guiHeight / 2);
+		
+		this.blit(matrixStack, (this.width / 2) - (guiWidth / 2), yOffset, 0, 0, guiWidth, guiHeight);
+		font.draw(matrixStack, divideValueMessage, (width / 2) - font.width(divideValueMessage) / 2, yOffset + 40, 0x404040);
+		super.render(matrixStack, mouseX, mouseY, partialTicks);
+	}
+	
+	private void finish()
+	{
+		StatStorerPacket packet = new StatStorerPacket(activeType, te.getBlockPos(), textToInt());
+		MSPacketHandler.sendToServer(packet);
+		onClose();
+	}
+	
+	private int textToInt()
+	{
+		try
+		{
+			divideValueBy = Integer.parseInt(divideTextField.getValue());
+		} catch(NumberFormatException ignored)
+		{
+		}
+		
+		return divideValueBy;
```

Comment:
As a way to update divideValueBy before it is sent to the packet, without adding an extra line to finish()

Reply:
I'm not sure I follow. Can you elaborate?

Reply:
Perhaps there is something redundant or unnecessary about the way I am doing it, but I use textToInt() as a means of collecting the new divideValueBy from divideTextField while checking that it is viable. This is in order to avoid adding another line in the finish function to make this update apply to divideValueBy

Reply:
But as far as I can see, `divideValueBy` isn't used anywhere other than in this function. Why not just have `textToInt()` return the parsed value directly?

Reply:
I see what you mean now, I was viewing the problem from another perspective!

Reply:
Well, what do you want to happen if it catches an exception there?

Reply:
I am not sure, but I was under the assumption that the catch was necessary for the try component to work

Reply:
Indeed, catch is necessary while using try

Reply:
Then could I removed the textToInt function and just put "Integer.parseInt(divideTextField.getValue())" straight into StatStorerPacket?

Reply:
Since the text field can accept non integer forming characters, it needs to be able to catch bad inputs otherwise it causes a crash

Reply:
An exception from parsing an int here is normal and expected behavior, so it should definitely be caught and handled in some way. The question is then: how should it be handled?
Right now, when it tries to parse something that isn't a number, it returns `divideValueBy`, which is the last value that was successfully parsed in `textToInt()`, or 0 if it hasn't managed to parse a value yet. This doesn't seem right.
One idea is to write `finish()` such that it does not send a packet if it cannot parse a number from the text field.